src/Columns/ColumnString.cpp

```diff
@@ -274,15 +274,8 @@ void ColumnString::collectSerializedValueSizes(PaddedPODArray<UInt64> & sizes, c
     {
         for (size_t i = 0; i < rows; ++i)
         {
-            if (is_null[i])
-            {
-                ++sizes[i];
-            }
-            else
-            {
-                size_t string_size = sizeAt(i);
-                sizes[i] += sizeof(string_size) + string_size + 1 /* null byte */;
-            }
+            size_t string_size = sizeAt(i);
+            sizes[i] += !!is_null[i] + !is_null[i] * (sizeof(string_size) + string_size + 1 /* null byte */);
```

Comment:
Does this exact change help?

Reply:
`ColumnString::collectSerializedValueSizes` takes about 10% of overall cpu time. I guess the changes should be helpful.